backend/app/services/implementations/meal_request_service.py

```diff
@@ -26,16 +29,59 @@ def create_meal_request(
                     description=description,
                     requestor=requestor,
                     meal_info=meal_info,
-                    donation_datetime=datetime.combine(request_date, drop_off_time),
+                    drop_off_datetime=datetime.combine(request_date, drop_off_time),
                     drop_off_location=drop_off_location,
                     delivery_instructions=delivery_instructions,
                     onsite_staff=onsite_staff,
                 )
                 new_meal_request.save()
-                print(new_meal_request.to_serializable_dict())
                 meal_requests.append(new_meal_request)
         except Exception as error:
             self.logger.error(str(error))
             raise error
 
         return map(lambda x: x.to_serializable_dict(), meal_requests)
+
+    def get_meal_requests_by_requestor_id(
+        self,
+        requestor_id,
+        min_drop_off_date,
+        max_drop_off_date,
+        status,
+        offset,
+        limit,
+        sort_by_date_direction,
+    ):
+        try:
+            sort_prefix = "+"
+            if sort_by_date_direction == SortDirection.DESCENDING:
+                sort_prefix = "-"
+
+            requests = MealRequest.objects(
+                requestor=requestor_id,
+                status__in=status,
+            ).order_by(f"{sort_prefix}date_created")
+
+            if min_drop_off_date is not None:
+                requests = requests.filter(
+                    drop_off_datetime__gte=min_drop_off_date,
+                )
+            if max_drop_off_date is not None:
+                requests = requests.filter(
+                    drop_off_datetime__lte=max_drop_off_date,
+                )
+            if limit is not None:
+                requests = requests[offset : offset + limit]
+            else:
+                requests = requests[offset:]
+
+            meal_request_dtos = []
+            for request in requests:
+                request_dict = request.to_serializable_dict()
+                meal_request_dtos.append(MealRequestDTO(**request_dict))
+
```

Comment:
When I first read this, I thought that you were first fetching all the meal requests, and then doing the filtering (which we don't want to do, since that would be super slow). But from what I read on the docs, it seems that mongoengine only actually gets the data from the database once you actually iterate through  the `requests` in the for loop, so that means this query fine. Maybe leaving a quick comment here would be helpful for future readers?

https://docs.mongoengine.org/guide/querying.html#

Reply:
That's cool, I didn't know that actually! 

My hope was to make it so that the **minimum** amount of required data is fetched from the database before filtering, to reduce unnecessary fetching. The problem is that the `min_drop_off_date`, `max_drop_off_date`, and `limit` params can be unspecified, and I couldn't figure out a way to structure the initial mongoengine query so that it filters conditionally (e.g. `filter for results where drop_off_date >= min_drop_off_date only if min_drop_off_date is specified`). If there is a good way to do this in the initial query, I would love to know! But since we're not worried about scale at this point, I don't the performance of this query should cause any issues.

Reply:
For sure, I can leave a comment 👍